src/server/kiwix-serve.cpp

```diff
@@ -324,18 +324,12 @@ int main(int argc, char** argv)
   auto libraryFileTimestamp = newestFileTimestamp(libraryPaths);
   auto curLibraryFileTimestamp = libraryFileTimestamp;
 
-  /* Infer ipMode from address */
-  kiwix::IpMode ipMode = kiwix::IpMode::ipv4;
-
-  if (address == "all"){
-    address = "";
-    ipMode = kiwix::IpMode::all;
-  } else if (address == "ipv4"){
-    address = "";
-  } else if (address == "ipv6"){
-    address = "";
-    ipMode = kiwix::IpMode::ipv6;
-  }
+  kiwix::IpMode ipMode =  (address == "all")  ? kiwix::IpMode::ALL  :
+                          (address == "ipv6") ? kiwix::IpMode::IPV6 :
+                          (address == "ipv4") ? kiwix::IpMode::IPV4 :
+                                                kiwix::IpMode::AUTO;
+
+  if(!address.empty() && ipMode != kiwix::IpMode::AUTO) address.clear();  // Clear to indicate that protocol has been provided in address.
```

Comment:
Long lines are not good.

Reformat as a multiline statement with braces (like it was in the previous iteration).